### PR TITLE
Crater Run: Make tyvar_behind_raw_pointer an error

### PR DESCRIPTION
**Pull request by mikeyhew**
In preparation for a crater run to determine its impact

### CODE REVIEW

File: src/librustc/lint/builtin.rs [hunk 1]
```diff
@@ -242,7 +242,7 @@ declare_lint! {
 
 declare_lint! {
     pub TYVAR_BEHIND_RAW_POINTER,
-    Warn,
+    Deny,
```
Comment (petrochenkov) [inline src/librustc/lint/builtin.rs]: `Forbid` is better for testing existing lints, someone could `allow` it already.

**Comment by mikeyhew** [comment]
@petrochenkov good point, and some crates could do `allow(warnings)`, right?

**Comment by kennytm** [comment]
@aidanhs @Mark-Simulacrum crater run wanted.

**Comment by Mark-Simulacrum** [comment]
Crater run started.

**Comment by Mark-Simulacrum** [comment]
http://cargobomb-reports.s3.amazonaws.com/pr-46934/index.html (yes, the pr-# is wrong, but this should be the right run -- if not, let me know, we'll restart and rerun it).

**Comment by mikeyhew** [comment]
60 regressions – is that good? It's better than I expected, anyway. And is it safe to say that all of the "fixed" crates and any test-fail regressions were spurious?

@nikomatsakis what do the results tell you?

**Comment by nikomatsakis** [comment]
60 seems like a lot to me =) but it all depends on the details. Let me dig in a bit.

**Comment by nikomatsakis** [comment]
I got as far as `link-0.1.0`. Most of the results I've looked at look like legit, distinct regressions. Here are the ones I see that are spurious:

- cc-1.0.4
- conc-0.5.0
- crux-0.3.0
- ctx-0.2.0
- esprit-0.0.5
- gear-0.1.3
- hashconsing-0.3.0
- hornet-0.1.0 
- kernel_density-0.0.1
- libfuzzy-sys-0.1.0

This merits some discussion and consideration, I think. Of course, the future compatibility warning may lead some people to migrate, but we may want to think about some special method resolution rules -- for example, giving inherent methods defined on `*mut T` some sort of special precedence.

**Comment by nikomatsakis** [comment]
Well, I think this PR has served its purpose, right? I'm going to close for now. (It's not meant to be merged, I don't think.)

**Pull request closed without merging**